src/Control/Monad/Catch.hs

```diff
@@ -179,6 +178,21 @@ class MonadCatch m => MonadMask m where
   -- and/or unkillable.
   uninterruptibleMask :: ((forall a. m a -> m a) -> m b) -> m b
 
+  -- | A generalized version of the standard bracket function which
+  -- allows distinguishing different exit cases.
+  --
+  -- @since 0.8.4
```

Comment:
Yes, good catch, I had a previous attempt that included backwards compat that ultimately fell through.